### PR TITLE
bump cli to use buildkit 0.17

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

- Bumps Buildkit version to match 0.17
- Move getSolveOpt to buildkit package as it's only used there
- Moves authprovider to buildkit package as it's used with the solveOpt
- Adds the frontend version based on the client and check if we can use the local frontend or the gateway

## How to validate

Please provide step-by-step instructions to replicate your validation scenario. For bug fixes, detail how to reproduce both the bug and its fix, along with any observations.

1. Run okteto deploy --remote
1. Run okteto build
1.

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: pkg/build/buildkit/opt.go [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+					"ref":  exportCacheTo,
+					"mode": "max",
+				},
+			},
+		)
+	}
+	// TODO(#3548): remove when we upgrade buildkit to 0.11
+	if len(opt.CacheExports) > 1 {
+		opt.CacheExports = opt.CacheExports[:1]
+	}
```
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply pkg/build/buildkit/opt.go:288]: If we want to maintain retrocompatibility we should move it inside the GetFrontend and use gateway so we can export several

File: pkg/build/buildkit/opt.go [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+	if err != nil {
+		logger.Infof("failed to get buildkit info: %s", err)
+		return nil, err
+	}
+	buildkitVersion, err := semver.NewVersion(info.BuildkitVersion.Version)
+	if err != nil {
+		logger.Infof("failed to parse buildkit version: %s", err)
+		return nil, err
+	}
+	// We substract 6 as it's the difference between the buildkit version and the docker frontend version
```
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply pkg/build/buildkit/opt.go:100]: I checked all the versions to 0.9 and it was true on all of them

File: pkg/build/buildkit/opt.go [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+			token:    b.okCtx.GetCurrentToken(),
+			cert:     b.okCtx.GetCurrentCertStr(),
+		}
+
+		ap := newDockerAndOktetoAuthProvider(b.okCtx.GetRegistryURL(), b.okCtx.GetCurrentUser(), b.okCtx.GetCurrentToken(), apCtx, os.Stderr)
+		attachable = append(attachable, ap)
+	} else {
+		dockerCfg := dockerConfig.LoadDefaultConfigFile(os.Stderr)
+		attachable = append(attachable, authprovider.NewDockerAuthProvider(dockerCfg, map[string]*authprovider.AuthTLSConfig{}))
+	}
```
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply pkg/build/buildkit/opt.go:209]: Created https://okteto.atlassian.net/browse/DEV-805

File: pkg/build/buildkit/opt.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply pkg/build/buildkit/opt.go]: Done in https://github.com/okteto/okteto/pull/4562/commits/023021992ae159b8256052d53d89c6d31c9ce2dd

File: pkg/build/buildkit/opt.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ifbyol) [reply pkg/build/buildkit/opt.go:100]: Yeah, agree with doing it based on the BuildKit version.

**Review by ifbyol — APPROVED** [review]

**Pull request merged**